File: upload_to_firebase.py
import json
import urllib.request
import urllib.error

# Firebase URL
FIREBASE_URL = "https://tac-production-bfd08-default-rtdb.asia-southeast1.firebasedatabase.app/INV_PRODUCTS.json"

# Load new master data
try:
    with open('data/new_master.json', 'r', encoding='utf-8') as f:
        data = json.load(f)
except Exception as e:
    print(f"Error loading JSON: {e}")
    exit(1)

# Upload via REST API (PUT overwrites the node)
req = urllib.request.Request(FIREBASE_URL, method='PUT')
req.add_header('Content-Type', 'application/json')
json_data = json.dumps(data).encode('utf-8')

try:
    with urllib.request.urlopen(req, data=json_data) as response:
        if response.status == 200:
            print("Successfully uploaded to Firebase!")
        else:
            print(f"Failed with status: {response.status}")
            print(response.read().decode('utf-8'))
except urllib.error.URLError as e:
    print(f"URLError: {e}")

File: upload_to_firebase.js
const fs = require('fs');
const { initializeApp } = require('firebase/app');
const { getDatabase, ref, set } = require('firebase/database');

const firebaseConfig = {
    apiKey: "AIzaSyAtXI3uz70l5P6UF26OC1Tru1fte35343g",
    authDomain: "tac-production-bfd08.firebaseapp.com",
    databaseURL: "https://tac-production-bfd08-default-rtdb.asia-southeast1.firebasedatabase.app",
    projectId: "tac-production-bfd08",
    storageBucket: "tac-production-bfd08.firebasestorage.app",
    messagingSenderId: "299994554225",
    appId: "1:299994554225:web:33f0de3f5f9e53f21de0d0"
};

const app = initializeApp(firebaseConfig);
const db = getDatabase(app);

const rawData = fs.readFileSync('data/new_master.json', 'utf8');
const data = JSON.parse(rawData);

set(ref(db, 'INV_PRODUCTS'), data).then(() => {
    console.log("Successfully uploaded!");
    process.exit(0);
}).catch((err) => {
    console.error("Error uploading:", err);
    process.exit(1);
});
